plaso/parsers/sqlite_plugins/chrome_cookies.py

```diff
@@ -54,11 +54,18 @@ class ChromeCookiePlugin(interface.SQLitePlugin):
 
   # Define the needed queries.
   QUERIES = [
+      # Query for Chrome versions 17 - 65.
       (('SELECT creation_utc, host_key, name, value, path, expires_utc, '
         'secure, httponly, last_access_utc, has_expires, persistent '
+        'FROM cookies'), 'ParseCookieRow'),
+      # Query for Chrome versions 66 and above. The column names changed
+      # slightly, but the values are the same.
+      (('SELECT creation_utc, host_key, name, value, path, expires_utc, '
```

Comment:
This means there's going to be an error logged with "unable to run query", when there isn't really a problem. Is there any way around that? Alter the queries, or check for a different table?

Reply:
we could use CASE WHEN (https://sqlite.org/syntax/expr.html) but it seems sqlite check that all columns names are present in the table. otherwise, we would have to split the queries in two classes.

Reply:
Yeah, this is what I meant by not ideal. The table check doesn't help here, as the table names are the same. The only way I can see to avoid this error is to split the plugin into multiple sub-plugins that run on each version of the database and enforce an exact schema match. The downside to this is that it makes it "fragile"; minor changes that don't effect the data we're parsing (like adding an additional column or renaming a column we don't even look at) will break the plugin. 

In my opinion, the "required tables" check is too coarse, and the "exact schema match" is too fine. It would be great to have an in-between option, something like "required columns", which would function like "required tables" but one step lower. If the columns specified in "required columns" are a subset of what's present, then run the plugin. I think this approach would help with a bunch of plugins, as the schemas of many databases are constantly evolving. 

I have more ideas about this but it might be a bit much for this PR. I can make an issue for this and we can talk about it more? And in the meantime just make the Cookies parser use the schema matching? Let me know what you think.

Reply:
Fair enough - I think we can live with the error message here for now. I think it's clear we need some of "partial schema match" functionality. I've opened #2133 to track that. I'll merge this for now though.